From 8c2aaa775f9385c75600d22e4bb31f82d8082db5 Mon Sep 17 00:00:00 2001
From: Juri Lelli <juri.lelli@arm.com>
Date: Fri, 6 Mar 2015 11:01:28 +0000
Subject: [PATCH 2/3] DEBUG: add cpu to load_avg_task tracepoint

This is useful when plotting where a task was running during an
experiment.

adapted for 4.3-rc1 (PELT rewrite problem)

Change-Id: I5a90aaf3eb1254b4d1c6ec3d7e4159d9643b0b71
Signed-off-by: Juri Lelli <juri.lelli@arm.com>
Signed-off-by: Koan-Sin Tan <freedom.tan@linaro.org>
---
 include/trace/events/sched.h | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/include/trace/events/sched.h b/include/trace/events/sched.h
index 086d716..a82f6c9 100644
--- a/include/trace/events/sched.h
+++ b/include/trace/events/sched.h
@@ -599,6 +599,7 @@ TRACE_EVENT(sched_load_avg_task,
 	TP_STRUCT__entry(
 		__array( char,	comm,	TASK_COMM_LEN		)
 		__field( pid_t,	pid				)
+		__field( int,	cpu				)
 		__field( unsigned long,	last_update_time	)
 		__field( unsigned long,	load_sum		)
 		__field( unsigned int,	util_sum		)
@@ -610,6 +611,7 @@ TRACE_EVENT(sched_load_avg_task,
 	TP_fast_assign(
 		memcpy(__entry->comm, tsk->comm, TASK_COMM_LEN);
 		__entry->pid			= tsk->pid;
+		__entry->cpu			= task_cpu(tsk);
 		__entry->last_update_time	= avg->last_update_time;
 		__entry->load_sum		= avg->load_sum;
 		__entry->util_sum		= avg->util_sum;
@@ -618,12 +620,12 @@ TRACE_EVENT(sched_load_avg_task,
 		__entry->util_avg		= avg->util_avg;
 	),
 
-	TP_printk("comm=%s pid=%d last_update=%lu load_sum=%lu util_sum=%u "
-		  "period_contrib=%u load_avg=%lu util_avg=%lu",
-		  __entry->comm, __entry->pid, __entry->last_update_time,
-		  __entry->load_sum, __entry->util_sum,
-		  __entry->period_contrib,
-		  __entry->load_avg, __entry->util_avg)
+	TP_printk("comm=%s pid=%d cpu=%d last_update=%lu load_sum=%lu "
+		  "util_sum=%u period_contrib=%u load_avg=%lu util_avg=%lu",
+		__entry->comm, __entry->pid, __entry->cpu,
+		__entry->last_update_time, __entry->load_sum,
+		__entry->util_sum, __entry->period_contrib,
+		__entry->load_avg, __entry->util_avg)
 );
 
 /*
-- 
1.9.1

